src/file/flock.cr

```diff
@@ -0,0 +1,58 @@
+lib LibC
+  @[Flags]
+  enum FlockOp
+    SH = 0x1
+    EX = 0x2
+    NB = 0x4
+    UN = 0x8
+  end
+
+  fun flock(fd : Int, op : FlockOp) : Int
+end
+
+class File
+  def flock_shared blocking = true
+    flock_shared blocking
+    begin
+      yield
+    ensure
+      flock_unlock
+    end
+  end
+
+  # Place a shared advisory lock. More than one process may hold a shared lock for a given file at a given time.
+  # Errno::EWOULDBLOCK is raised if blocking = false and an existing exclusive lock is set.
```

Comment:
I would prefer "if \*blocking\* is set to \`false\`", people already try to call with keyword args using `foo(bar=baz)`